### PR TITLE
libxml2 2.13.1

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4629](https://anaconda.atlassian.net/browse/PKG-4629)
- [Upstream repository](https://gitlab.gnome.org/GNOME/libxml2/-/tree/2.13?ref_type=heads)
- [Upstream changelog/diff](https://gitlab.gnome.org/GNOME/libxml2/-/releases)

### Explanation of changes:

- Set version and sha256
- Update `build.sh`
- Remove outdated CVE patch

[PKG-4629]: https://anaconda.atlassian.net/browse/PKG-4629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ryanskeith — APPROVED** [review]
LGTM

File: recipe/build.sh [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/bin/bash
 
+export NOCONFIGURE=1
```
Comment (ryanskeith) [inline recipe/build.sh:3]: This is interesting.  Why is it set to noconfigure?

File: recipe/build.sh [hunk 2]
```diff
@@ -15,13 +16,7 @@ export CXXFLAGS="${CXXFLAGS} -DTRUE=1"
...
             --enable-static=no
 make -j${CPU_COUNT} ${VERBOSE_AT}
-# Sorry:
-# ## Error cases regression tests
-# file result/errors/759573.xml.err is 1983 bytes, result is 1557 bytes
-# Error for ./test/errors/759573.xml failed
-# if [[ ${target_platform} != osx-64 ]]; then
-#   make check $VERBOSE_AT}
-# fi
+
```
Comment (cbouss) [inline recipe/build.sh:19]: Can you try adding make check?

File: recipe/build.sh (hunk 1 above)
Comment (cbouss) [reply recipe/build.sh:3]: Please add a comment

**Review by cbouss — COMMENTED** [review]
Reading [NEWS · master · GNOME / libxml2 · GitLab](https://gitlab.gnome.org/GNOME/libxml2/-/blob/master/NEWS)  --with-legacy should be used for maximum compatibility.

File: recipe/build.sh (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/build.sh:3]: > `./autogen.sh` will run `configure` unless you tell it not too... Alternatively we could pass all the configure flags to `./autogen.sh` which will pass them to the configure script.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
   name: libxml2
   version: {{ version }}
 
 source:
   url: https://gitlab.gnome.org/GNOME/libxml2/-/archive/v{{ version }}/libxml2-v{{ version }}.tar.gz
-  sha256: 1aa47bd54f9e0245686d494fbbbfa4e3e77b6fc4f988708383de8a1033292e66
+  sha256: d1d90c6cecedd4a572af7aef6606bc7a1b38bcc09deef182dd065685fbd8de3f
   patches:
     - 0002-Make-and-install-a-pkg-config-file-on-Windows.patch  # [win]
-    - 0003-CVE-2024-25062.patch
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:12]: Pleas remove the file from the repo too.

File: recipe/build.sh (hunk 1 above)
Comment (SC426) [reply recipe/build.sh:4]: Comment has been added

File: recipe/meta.yaml (hunk 3 above)
Comment (SC426) [reply recipe/meta.yaml:12]: Done

File: recipe/build.sh [hunk 4]
```diff
@@ -13,15 +15,14 @@ export CXXFLAGS="${CXXFLAGS} -DTRUE=1"
...
+if [[ ${target_platform} != osx-64 ]]; then
+  make check $VERBOSE_AT}
```
Comment (JeanChristopheMorinPerso) [inline recipe/build.sh:23]:
```suggestion
  make check "${VERBOSE_AT}"
```

Comment (SC426) [reply recipe/build.sh]: Fixed!

**Review by JamesRobertsonGames — APPROVED** [review]

**Pull request merged**